lib/Raven/SerializerInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+
+namespace Raven;
+
+interface SerializerInterface
+{
+    /**
+     * Serialize an object (recursively) into something safe for data
+     * sanitization and encoding.
+     *
+     * @param mixed $value
+     * @param int   $max_depth
+     * @param int   $_depth
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, $max_depth = 3, $_depth = 0);
```

Comment:
> should we provide a basic implementation of a serializer

We did it already. \\Raven\\Serializer is it. Anyone can extend it

> make the Serializer::serializeObject protected
> change the signature of 

I will do it tomorrow. 

> it was a really big mistake to have all properties/methods public

Okay. Which of us will make a pull request that will make all methods protected?